From fb6406c5fed0781a69b39550894d2d70d2ff3446 Mon Sep 17 00:00:00 2001
From: Pascal Paillet <p.paillet@foss.st.com>
Date: Tue, 13 Aug 2024 17:32:49 +0200
Subject: [PATCH 1080/1695] dts: stm32: add GPU OPP description

Add the description for GPU operating points.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 7f7186808936b81f83b4f93addfe564ae13eceaf dts: stm32: add GPU OPP description in stm32mp235.dtsi
d8e84f2e09ec7fe722aa3184663dec378eca4e5e dts: stm32: add GPU OPP description]
Change-Id: I2f9fe4df3745b48df33c1e08a0da139072b368c8
Signed-off-by: Pascal Paillet <p.paillet@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/398312
Tested-by: Pascal PAILLET <p.paillet@st.com>
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Pascal PAILLET <p.paillet@st.com>
---
 core/arch/arm/dts/stm32mp255.dtsi | 27 +++++++++++++++++++++++++++
 1 file changed, 27 insertions(+)

diff --git a/core/arch/arm/dts/stm32mp255.dtsi b/core/arch/arm/dts/stm32mp255.dtsi
index e6fa59621..1c221ead6 100644
--- a/core/arch/arm/dts/stm32mp255.dtsi
+++ b/core/arch/arm/dts/stm32mp255.dtsi
@@ -6,4 +6,31 @@
 #include "stm32mp253.dtsi"
 
 / {
+	gpu_opp_table: opp-table-gpu {
+		compatible = "operating-points-v2";
+
+		opp-800000000 {
+			opp-hz = /bits/ 64 <800000000>;
+			opp-microvolt = <800000>;
+			opp-supported-hw = <0x3>;
+			st,opp-default;
+		};
+
+		opp-900000000 {
+			opp-hz = /bits/ 64 <900000000>;
+			opp-microvolt = <900000>;
+			opp-supported-hw = <0x2>;
+			st,opp-default;
+		};
+	};
+};
+
+&rifsc {
+	gpu: gpu@48280000 {
+		compatible = "vivante,gc";
+		gpu-supply = <&vddgpu>;
+		clocks = <&rcc CK_BUS_GPU>;
+		operating-points-v2 = <&gpu_opp_table>;
+		access-controllers = <&rifsc STM32MP25_RIFSC_GPU_ID>;
+	};
 };
-- 
2.47.2

